File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        if not prices:
            return 0 

        min_price = float("inf")
        max_profit = 0

        for i in range(len(prices)):
            max_profit = max((prices[i] - min_price), max_profit)
            min_price = min(prices[i], min_price)

        return max_profit

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        
        int profit = 0;
        int lowestPrice = Integer.MAX_VALUE;

        for (int price : prices){
            if (price < lowestPrice){
                lowestPrice = price;
            }
            profit = Math.max(profit, price - lowestPrice);
            
        }
        
        return profit;

    }
}